src/query/service/src/interpreters/interpreter_insert_v2.rs

```diff
@@ -177,6 +137,93 @@ impl InsertInterpreterV2 {
         let cast_needed = select_schema != *output_schema;
         Ok(cast_needed)
     }
+
+    async fn schedule_insert_select(
+        &self,
+        plan: &Plan,
+        table: Arc<dyn Table>,
+    ) -> Result<SendableDataBlockStream> {
+        // select_plan is already distributed optimized
+        let (mut select_plan, select_column_bindings) = match plan {
+            Plan::Query {
+                s_expr,
+                metadata,
+                bind_context,
+                ..
+            } => {
+                let builder = PhysicalPlanBuilder::new(metadata.clone(), self.ctx.clone());
+                (builder.build(s_expr).await?, bind_context.columns.clone())
+            }
+            _ => unreachable!(),
+        };
+
+        table.get_table_info();
+        let catalog = self.plan.catalog.clone();
+        let is_distributed_plan = select_plan.is_distributed_plan();
+
+        let insert_select_plan = match select_plan {
+            PhysicalPlan::Exchange(ref mut exchange) => {
```

Comment:
So aggregation node will not go into this path ? cc @zhang2014

Reply:
> So aggregation node will not go into this path ? cc @zhang2014

It will and can be covered by this cases.

Reply:
For example:

```
mysql> explain select sum(a), sum(b) from n2;
+----------------------------------------------------------------------+
| explain                                                              |
+----------------------------------------------------------------------+
| Project                                                              |
| ├── columns: [sum(a) (#2), sum(b) (#3)]                              |
| └── EvalScalar                                                       |
|     ├── expressions: [sum(b) (#5), sum(a) (#4)]                      |
|     └── AggregateFinal                                               |
|         ├── group by: []                                             |
|         ├── aggregate functions: [sum(a), sum(b)]                    |
|         └── Exchange                                                 |
|             ├── exchange type: Merge                                 |
|             └── AggregatePartial                                     |
|                 ├── group by: []                                     |
|                 ├── aggregate functions: [sum(a), sum(b)]            |
|                 └── TableScan                                        |
|                     ├── table: default.default.n2                    |
|                     ├── read rows: 20000                             |
|                     ├── read bytes: 160282                           |
|                     ├── partitions total: 4                          |
|                     ├── partitions scanned: 4                        |
|                     └── push downs: [filters: [], limit: NONE]       |
+----------------------------------------------------------------------+
19 rows in set (0.09 sec)
Read 0 rows, 0.00 B in 0.024 sec., 0 rows/sec., 0.00 B/sec.

mysql> explain select a from n2 group by a;
+------------------------------------------------------------------+
| explain                                                          |
+------------------------------------------------------------------+
| Exchange                                                         |
| ├── exchange type: Merge                                         |
| └── AggregateFinal                                               |
|     ├── group by: [a]                                            |
|     ├── aggregate functions: []                                  |
|     └── Exchange                                                 |
|         ├── exchange type: Hash(_group_by_key)                   |
|         └── AggregatePartial                                     |
|             ├── group by: [a]                                    |
|             ├── aggregate functions: []                          |
|             └── TableScan                                        |
|                 ├── table: default.default.n2                    |
|                 ├── read rows: 20000                             |
|                 ├── read bytes: 80138                            |
|                 ├── partitions total: 4                          |
|                 ├── partitions scanned: 4                        |
|                 └── push downs: [filters: [], limit: NONE]       |
+------------------------------------------------------------------+
17 rows in set (0.08 sec)
Read 0 rows, 0.00 B in 0.024 sec., 0 rows/sec., 0.00 B/sec.
```

Reply:
ok